### PR TITLE
Spring Boot 2.1 Migration

### PR DESCRIPTION
**Pull request by stefbehl**
Migrate hawkBit examples (e.g. device simulator) to Spring Boot 2.1 (and Spring 5)

### CODE REVIEW

File: hawkbit-example-ddi-feign-client/pom.xml [hunk 1]
```diff
@@ -31,25 +31,9 @@
...
-         <groupId>com.fasterxml.jackson.core</groupId>
-         <artifactId>jackson-databind</artifactId>
-      </dependency>
-      <dependency>
-         <groupId>org.hibernate</groupId>
-         <artifactId>hibernate-validator</artifactId>
-      </dependency>
+         <groupId>org.springframework.cloud</groupId>
+         <artifactId>spring-cloud-starter-openfeign</artifactId>
+      </dependency>
```
Comment (stefbehl) [inline hawkbit-example-ddi-feign-client/pom.xml]: check if validator is needed

File: hawkbit-example-mgmt-feign-client/pom.xml [hunk 2]
```diff
@@ -31,24 +31,8 @@
...
-      </dependency>
-      <dependency>
-         <groupId>com.fasterxml.jackson.core</groupId>
-         <artifactId>jackson-databind</artifactId>
-      </dependency>
-      <dependency>
-         <artifactId>hibernate-validator</artifactId>
-         <groupId>org.hibernate</groupId>
+         <groupId>org.springframework.cloud</groupId>
+         <artifactId>spring-cloud-starter-openfeign</artifactId>
```
Comment (stefbehl) [inline hawkbit-example-mgmt-feign-client/pom.xml]: check if validator is needed

File: pom.xml [hunk 3]
```diff
@@ -46,9 +46,9 @@
    </modules>
 
    <properties>
-      <feign.version>9.7.0</feign.version>
-      <feign.extension.version>9.7.0</feign.extension.version>
-      <hawkbit.version>0.2.4</hawkbit.version>
+      <spring.feign.version>2.0.2.RELEASE</spring.feign.version>
```
Comment (stefbehl) [inline pom.xml]: check if version override is still needed

File: pom.xml [hunk 4]
```diff
@@ -189,30 +189,15 @@
             <version>${hawkbit.version}</version>
          </dependency>
          <dependency>
-            <groupId>io.github.openfeign</groupId>
-            <artifactId>feign-core</artifactId>
-            <version>${feign.version}</version>
+            <groupId>org.springframework.cloud</groupId>
```
Comment (stefbehl) [inline pom.xml]: can probably be removed (see above)

**Comment by StefanKlt** [comment]
👍

**Pull request merged**